controllers/link.go

```diff
@@ -0,0 +1,63 @@
+package controllers
+
+import (
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	"github.com/nimble-link/backend/models"
+	"github.com/nimble-link/backend/pkg/linkutils"
+	"github.com/nimble-link/backend/services/authentication"
+)
+
+type ShortLinkInput struct {
+	OriginalURL string `json:"original_url" binding:"required"`
+	CustomAlias string `json:"custom_alias"`
+	Password    string `json:"password"`
+}
+
+func CreateLink(c *gin.Context) {
+
+	var input ShortLinkInput
+	if err := c.ShouldBindJSON(&input); err != nil {
+		c.AbortWithStatus(http.StatusBadRequest)
+		return
+	}
+
+	// get user info
+	user, err := authentication.GetCurrentUserFromContext(c)
+	if err != nil {
+		c.AbortWithStatus(http.StatusBadRequest)
+		return
+	}
+
+	userID := user.ID
+	originalURL := input.OriginalURL
+
+	if customAlias := input.CustomAlias; customAlias != "" {
+		// save user's CustomAlias
+		if linkutils.IsDuplicateCustomAlias(customAlias) {
+			c.JSON(http.StatusBadRequest, gin.H{"error": "Duplicate Custom Alias"})
+		} else {
+			saveLink(c, originalURL, customAlias, input.Password, userID)
+		}
+	} else {
+		// generate random CustomAlias
+		customAlias := linkutils.GenerateHashFrom(originalURL, userID)
+
+		for linkutils.IsDuplicateCustomAlias(customAlias) {
```

Comment:
Is it `if` instead of `for`?

Reply:
I want to use "while" instead of if in case we have duplicated custom alias multiple times :D

Reply:
got it. :D